system/gcov/gcov.c

```diff
@@ -69,6 +69,72 @@ static void gcov_dump(FAR const char * path, FAR const char *strip)
   __gcov_dump();
 }
 
+/****************************************************************************
+ * Name: stdout_dump
+ ****************************************************************************/
+
+#ifndef CONFIG_COVERAGE_TOOLCHAIN
+static void stdout_dump(FAR const void *buffer, size_t size,
+                        FAR void *arg)
+{
+  FAR const char *name = *((FAR const char **)arg);
+  struct lib_stdoutstream_s stdoutstream;
+  struct lib_hexdumpstream_s hexstream;
+  uint16_t checksum = 0;
+  int i;
+
+  if (size == 0)
+    {
+      return;
+    }
+
+  for (i = 0; i < size; i++)
+    {
+      checksum += ((FAR const uint8_t *)buffer)[i];
+    }
+
+  printf("gcov start filename:%s size: %zuByte\n", name, size);
+  lib_stdoutstream(&stdoutstream, stdout);
```

Comment:
why need initialize stream each time? let's move to gcov_stdout_dump

Reply:
Done